File: GFG/The Smurfs/The Smurfs.java

class Solution{
    static int minFind(int n, String a[]){
        // code here
        int r=0;
        int g=0;
        int b=0;
        for(int i=0;i<n;i++){
            if(a[i].equals("R")){
                r++;
            }else if(a[i].equals("G")){
                g++;
            }else{
                b++;
            }
        }
        if(r==n || g==n || b==n)return n;
        if((r%2==0 && b%2==0 && g%2==0) || (r%2!=0 && b%2!=0 && g%2!=0)){
            return 2;
        }
        return 1;
    }
}

File: GFG/The Smurfs/The Smurfs.cpp
class Solution{
public:
    int findMin(int n, char arr[]){
        // code here
        
          unordered_map<char,int>mp;
        for(int i = 0 ; i < n ; i++){
            mp[arr[i]]++;
        }
        int cnt = 0;
        for(auto x : mp){
            cnt+=x.second;
        }
       if(mp['R']%2==0 and mp['B']%2==0 and mp['G']%2==0){
           return 2;
       }
       else if(mp['R']%2==1 and mp['B']%2==1 and mp['G']%2==1){
           return 2;
       }
       else{
           return 1;
       }
    }
};
